text/0617-cloudfront-oac-l2.md

```diff
@@ -0,0 +1,805 @@
+
+### Using pre-existing S3 buckets
+
+If you are using an imported bucket for your S3 Origin and want to use OAC,
+you will need to update
+the S3 bucket policy manually. CDK apps cannot modify the configuration of imported constructs. After deploying the distribution, add the following
+policy statement to your
+S3 bucket to allow CloudFront read-only access
+(or additional permissions as required):
+
+```
+{
+    "Version": "2012-10-17",
+    "Statement": {
+        "Sid": "GrantOACAccessToS3",
+        "Effect": "Allow",
+        "Principal": {
+            "Service": "cloudfront.amazonaws.com"
+        },
+        "Action": "s3:GetObject",
+        "Resource": "arn:aws:s3:::<S3 bucket name>/*",
+        "Condition": {
+            "StringEquals": {
+                "AWS:SourceArn": "arn:aws:cloudfront::111122223333:distribution/<CloudFront distribution ID>"
+            }
+        }
+    }
+}
+```
+
+> Note: If your bucket previously used OAI, you will need to manually remove the policy statement
+that gives the OAI access to your bucket from your bucket policy.
+
+### Restricting access to an S3 Origin
+
+CloudFront provides two ways to send authenticated requests to an Amazon S3 origin:
+origin access control (OAC) and origin access identity (OAI).
+OAI is considered legacy due to limited functionality and regional
+limitations, whereas OAC is recommended because it supports All Amazon S3
+buckets in all AWS Regions, including opt-in Regions launched after
+December 2022, Amazon S3 server-side encryption with AWS KMS (SSE-KMS), and dynamic requests (PUT and DELETE) to Amazon S3. Additionally, OAC provides
+stronger security posture with short term credentials,
+and more frequent credential rotations as compared to OAI.
+(see [Restricting access to an Amazon S3 Origin](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/private-content-restricting-access-to-s3.html)).
+OAI and OAC can be used in conjunction with a bucket that is not public to
+require that your users access your content using CloudFront URLs and not S3 URLs directly.
+
+> Note: OAC and OAI can only be used with an regular S3 bucket origin (not a bucket configured as a website endpoint).
+
+To setup origin access control for an S3 origin, you can create an `S3OriginAccessControl`
+resource and pass it into the `originAccessControl` property of the origin:
+
+```ts
+const myBucket = new s3.Bucket(this, 'myBucket');
+const oac = new cloudfront.S3OriginAccessControl(this, 'myS3OAC');
+new cloudfront.Distribution(this, 'myDist', {
+  defaultBehavior: {
+    origin: new origins.S3BucketOriginWithOAC(myBucket, {
+      originAccessControl: oac
+    })
+  },
+});
+```
+
+If you use `S3BucketOriginWithOAC` and do not pass in an OAC, one will automatically be created for you and attached to the distribution.
+
+```ts
+const myBucket = new s3.Bucket(this, 'myBucket', {
+  objectOwnership: s3.ObjectOwnership.BUCKET_OWNER_ENFORCED,
```

Comment:
This is an implementation detail but we should check for the correct ownership or ACLs on buckets that are not imported and throw an error at synth if they are not properly setup.

Reply:
Noted!